app/src/main/kotlin/com/hippo/ehviewer/ui/main/AvatarIcon.kt

```diff
@@ -0,0 +1,165 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.NoAccounts
+import androidx.compose.material.icons.filled.Person
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.none
+import arrow.core.right
+import arrow.core.some
+import coil3.compose.AsyncImage
+import com.hippo.ehviewer.R
+import com.hippo.ehviewer.Settings
+import com.hippo.ehviewer.client.EhEngine
+import com.hippo.ehviewer.client.parser.HomeParser
+import com.hippo.ehviewer.collectAsState
+import com.hippo.ehviewer.ui.login.refreshAccountInfo
+import com.hippo.ehviewer.ui.tools.DialogState
+import com.hippo.ehviewer.util.displayString
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.retryWhen
+import kotlinx.coroutines.flow.shareIn
+import kotlinx.coroutines.launch
+import moe.tarsin.coroutines.runSuspendCatching
+
+private val limitScope = CoroutineScope(Dispatchers.IO)
+private val refreshEvent = MutableSharedFlow<Unit>()
+
+private val limitFlow = refreshEvent.map { EhEngine.getImageLimits().right() }
+    .retryWhen<Either<String, HomeParser.Result>> { cause, _ ->
+        emit(cause.displayString().left())
+        delay(15.seconds)
+        true
+    }.map { it.some() }.shareIn(limitScope, SharingStarted.Eagerly, replay = 1)
+
+context(CoroutineScope, DialogState, SnackbarHostState)
+@Composable
+fun AvatarIcon() {
+    val hasSignedIn by Settings.hasSignedIn.collectAsState()
+    val avatar by Settings.avatar.collectAsState()
+    if (hasSignedIn) {
+        val placeholder = stringResource(id = R.string.please_wait)
+        val resetImageLimitSucceed = stringResource(id = R.string.reset_limits_succeed)
+        val result by limitFlow.collectAsState(none())
+        LaunchedEffect(Unit) {
+            refreshEvent.emit(Unit)
+            refreshAccountInfo()
```

Comment:
Changing avatar is expected to be rare, so we should only refresh it if the user initiated it.

Reply:
Done